app/model/blockchain/token.py

```diff
@@ -199,33 +206,72 @@ def to_model(self) -> BondTokenModel:
         token_model.short_term_cache_created = datetime.utcnow()
         return token_model
 
-    def fetch_expiry_short(self) -> None:
+    async def fetch_expiry_short(self) -> None:
         """
         債権トークン リアルタイム属性情報取得
 
         :return: None
         """
-        token_contract = Contract.get_contract(
+        token_contract = AsyncContract.get_contract(
             TokenType.IbetStraightBond, self.token_address
         )
-
         # Fetch
-        owner_address = Contract.call_function(
-            token_contract, "owner", (), ZERO_ADDRESS
-        )
-        company = CompanyList.get_find(to_checksum_address(owner_address))
+        try:
+            async with SemaphoreTaskGroup(max_concurrency=3) as tg:
+                tasks = [
+                    tg.create_task(
+                        AsyncContract.call_function(
+                            token_contract, "owner", (), ZERO_ADDRESS
+                        )
+                    ),
+                    tg.create_task(
+                        AsyncContract.call_function(
+                            token_contract, "totalSupply", (), 0
+                        )
+                    ),
+                    tg.create_task(
+                        AsyncContract.call_function(token_contract, "status", (), True)
+                    ),
+                    tg.create_task(
+                        AsyncContract.call_function(
+                            token_contract, "transferable", (), False
+                        )
+                    ),
+                    tg.create_task(
+                        AsyncContract.call_function(
+                            token_contract, "isOffering", (), False
+                        )
+                    ),
+                    tg.create_task(
+                        AsyncContract.call_function(
+                            token_contract, "transferApprovalRequired", (), False
+                        )
+                    ),
+                    tg.create_task(
+                        AsyncContract.call_function(
+                            token_contract, "isRedeemed", (), False
+                        )
+                    ),
+                    tg.create_task(
+                        AsyncContract.call_function(token_contract, "memo", (), "")
+                    ),
+                ]
+            (
+                owner_address,
+                total_supply,
+                status,
+                transferable,
+                is_offering,
+                transfer_approval_required,
+                is_redeemed,
+                memo,
+            ) = [task.result() for task in tasks]
+        except ExceptionGroup:
+            raise ServiceUnavailable from None
```

Comment:
To asynchronously retrieve multiple contract attributes, using a task group.

Reply:
[Memo]
While `AsyncContract.call_function` can be called in the way like above, SQLAlchemy `AsyncSession` can not be called in task group because one async session can be used in only one task.

https://docs.sqlalchemy.org/en/20/orm/session_basics.html#session-faq-threadsafe